routes/memberProjects.js

```diff
@@ -88,4 +89,53 @@ router.get("/by-key/:associationId", async (req, res) => {
   });
 });
 
+router.get("/impersonate/:associationId", async (req, res) => {
+  const key = await req.ctx.db.collection("keys").findOne({
+    [req.associationType]: {$eq: req.params.associationId},
+  });
+
+  if (!key) {
+    res.status(404).json({message: "Not found"});
+    return;
+  }
+
+  const userDocument = await req.ctx.db.collection("users").findOne({
+    id: {$eq: key.userId},
+    disabled: {$ne: true},
+  });
+  if (!userDocument) {
+    res.status(404).json({message: "User not found or disabled"});
+    return;
+  }
+
+  let user;
+  try {
+    user = await findUser(key.userId);
```

Comment:
This pulls every attribute, which will be MUCH slower than just fetching attributes we want. Look at `findUser` in `routes/users.js`, notice how it only enumerates a few `attributes` we care about. I would make another function that just fetches a `uid` from a `ipaUniqueID`... It should be much faster

Reply:
Perhaps add an optional `attributes` parameter to `findUser`? Would be nice if that got moved out into another file because I was being lazy and stuck the same function in `routes/users.js` and `routes/memberProjects.js`